clickhouse/columns/array.h

```diff
@@ -260,8 +260,17 @@ class ColumnArrayT : public ColumnArray {
     using ColumnArray::Append;
 
     template <typename Container>
-    inline void Append(const Container& container) {
-        Append(std::begin(container), std::end(container));
+    inline void Append(Container&& container) {
+        using container_type = decltype(container);
+        if constexpr (std::is_lvalue_reference_v<container_type> || 
+            std::is_const_v<std::remove_reference_t<container_type>>) {
+            Append(std::begin(container), std::end(container));
+        }
+        else {
+            Append(std::make_move_iterator(std::begin(container)),
+                std::make_move_iterator(std::end(container)));
+            container.clear();
```

Comment:
IMO this is unnecessary, because after this operation container will be full of moved-out values. There is no way to spoil column data using that container or cause any other harm. If users want to clear container, they can do it after the call, in the user code.

Reply:
why add  container.clear() ?